### PR TITLE
Remove WithTelemetry option in favor of connect.WithInterceptors(otelconnect.NewInterceptor())

### PR DESCRIPTION
**Pull request by joshcarp**
The only use case we have of using this package is using a number of Interceptors instead of just the `WithTelemetry` option. It's anticipated that if a project is using connect-opentelemetry-go the chances are that it's probably also using other interceptors. In that case it makes sense to expose a `New` function that returns an `type interceptor struct` that can be plugged into the existing interceptor plumming. 

The only difference this makes to existing projects that are using `WithTelemetry` is that they now need to use `connect.WithInterceptor(otelconnect.New())`

Fixes: https://github.com/bufbuild/connect-opentelemetry-go/issues/21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,14 +39,14 @@ import (
...
-  // otelconnect.WithTelemetry adds tracing and metrics to both clients and
+  // otelconnect.New provides an interceptor that adds tracing and metrics to both clients and
   // handlers. By default, it uses OpenTelemetry's global TracerProvider and
   // MeterProvider, which you can configure by following the OpenTelemetry
   // documentation. If you'd prefer to avoid globals, use
   // otelconnect.WithTracerProvider and otelconnect.WithMeterProvider.
   mux.Handle(pingv1connect.NewPingServiceHandler(
     &pingv1connect.UnimplementedPingServiceHandler{},
-    otelconnect.WithTelemetry(),
+	  connect.WithInterceptors(otelconnect.New(),
```
Comment (doriable) [inline README.md:49]:
```suggestion
	  connect.WithInterceptors(otelconnect.New()),
```

File: README.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ func makeRequest() {
   client := pingv1connect.NewPingServiceClient(
     http.DefaultClient,
     "http://localhost:8080",
-    otelconnect.WithTelemetry(),
+    connect.WithInterceptors(otelconnect.New(),
```
Comment (doriable) [inline README.md:59]:
```suggestion
    connect.WithInterceptors(otelconnect.New()),
```

**Review by doriable — COMMENTED** [review]
This makes sense to me, left some nits for the README.
I'm also generally onboard with `New`, although I have a slight preference for `NewInterceptor` (but no strong opinions, happy with either).

**Review by akshayjshah — APPROVED** [review]
Two suggestions:

* If Doria leans toward `NewInterceptor`, that's fine with me - let's change.
* Let's export an `otelconnect.Interceptor` struct, and have the exported constructor return an `*Interceptor`. In general, let's return concrete types instead of interfaces (with the exception of constructors for functional options).

**Review by doriable — APPROVED** [review]

**Pull request merged**